networkx/algorithms/dag.py

```diff
@@ -139,6 +141,13 @@ def topological_sort(G, nbunch=None, reverse=False):
 
     if nbunch is None:
         nbunch = G.nodes_iter()
+        def direct_descendants(w):
+            return G[w]
+    else:
+        nbunch_index = {x: i for i, x in enumerate(nbunch)}
+        len_nbunch = len(nbunch)
+        def direct_descendants(w):
+            return sorted(G[w], key=lambda x: nbunch_index.get(x, len_nbunch))
     for v in nbunch:     # process all vertices in G
```

Comment:
In my brief search of topological sort descriptions, it is clear that people often create the sort starting only from a set of nodes that does not include the whole network. Looking at the history of the code I think this is the intent of nbunch -> to provide a starting set of nodes for the search. This should be better explained in the docstring. :}